Add example usage of `--head` option to `pr list` docs

babakks opened the pull request:
Fixes #10957 

This PR adds an example usage of the `--head` option to the `pr list` docs.

babakks committed:
Add example usage of `--head` option

Signed-off-by: Babak K. Shandiz <babakks@github.com>

babakks commented on pkg/cmd/pr/list/list.go:
```diff
@@ -64,6 +64,9 @@ func NewCmdList(f *cmdutil.Factory, runF func(*ListOptions) error) *cobra.Comman
 
+			# List PRs with a specific head branch name
+			$ gh pr list --head "typo"
+
```
That's actually the point, because currently we do not support such filters.

BagToad commented:
❓ What do you think about adding a note to the flag description to be more direct, instead of the example:

```Go
cmd.Flags().StringVarP(&opts.HeadBranch, "head", "H", "", "Filter by head branch. Does not support <owner>:<branch> syntax.")
```

babakks commented:
I thought about that, but since we're going to support the `<owner>:<branch>` syntax by #10945, I decided to keep it from changes in the near future.

What do you think?

BagToad commented:
> I thought about that, but since we're going to support the `<owner>:<branch>` syntax by #10945, I decided to keep it from changes in the near future.
> 
> What do you think?

I think this text would be very easy to remove and will help users understand in the meantime. Having only the example requires picking up on some nuance that by it being undocumented, it is unsupported.

Normally, I support this approach to not document all the things that _aren't_ possible. But in this case, we're clarifying an inconsistency with other commands that support the same flag but behave differently. So, I think it's appropriate to be very clear.

I'm also not sure how quickly someone would get to the work to add `<owner>:<branch>` support, so that is another reason why I support adding the text. If it truly was right away and before our next release, I think I wouldn't care as much. We can also make removing this text part of the AC for #10945, if we wanted to be thorough. 

Let me know if that makes sense!

babakks commented:
Makes sense, of course, with an A/C note on the other issue to update here accordingly. 

I'll update this PR soon.

babakks committed:
docs(pr list): mention `--head` does not support `<owner>:<branch>` syntax

Signed-off-by: Babak K. Shandiz <babakks@github.com>

babakks commented:
@BagToad I've updated this PR, and also the A/C in #10945 for future.

I changed the usage string to this because it felt a bit better without punctuations in the middle. What do you think?

```
Filter by head branch ("<owner>:<branch>" syntax not supported)
```

BagToad reviewed (approved):
Thanks! LGTM
